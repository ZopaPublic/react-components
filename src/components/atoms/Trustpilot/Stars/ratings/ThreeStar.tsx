import React from 'react';

export const THREE_STAR_ID = 'Trustpilot_ratings_ThreeStar';

export const ThreeStar = () => (
  <g id={THREE_STAR_ID} stroke="none" strokeWidth="1" fill="none" fillRule="evenodd">
    <g>
      <g>
        <rect id="Rectangle-path1" fill="#FFCE00" fillRule="nonzero" x="0" y="0" width="96" height="96"></rect>
        <rect id="Rectangle-path2" fill="#FFCE00" fillRule="nonzero" x="104" y="0" width="96" height="96"></rect>
        <rect id="Rectangle-path3" fill="#FFCE00" fillRule="nonzero" x="208" y="0" width="96" height="96"></rect>
        <rect id="Rectangle-path4" fill="#DCDCE6" fillRule="nonzero" x="312" y="0" width="96" height="96"></rect>
        <rect id="Rectangle-path5" fill="#DCDCE6" fillRule="nonzero" x="416" y="0" width="96" height="96"></rect>
        <path
          d="M48,64.7 L62.6,61 L68.7,79.8 L48,64.7 Z M81.6,40.4 L55.9,40.4 L48,16.2 L40.1,40.4 L14.4,40.4 L35.2,55.4 L27.3,79.6 L48.1,64.6 L60.9,55.4 L81.6,40.4 L81.6,40.4 L81.6,40.4 L81.6,40.4 Z"
          id="Shape1"
          fill="#FFFFFF"
          fillRule="nonzero"
        ></path>
        <path
          d="M152,64.7 L166.6,61 L172.7,79.8 L152,64.7 Z M185.6,40.4 L159.9,40.4 L152,16.2 L144.1,40.4 L118.4,40.4 L139.2,55.4 L131.3,79.6 L152.1,64.6 L164.9,55.4 L185.6,40.4 L185.6,40.4 L185.6,40.4 L185.6,40.4 Z"
          id="Shape2"
          fill="#FFFFFF"
          fillRule="nonzero"
        ></path>
        <path
          d="M256,64.7 L270.6,61 L276.7,79.8 L256,64.7 Z M289.6,40.4 L263.9,40.4 L256,16.2 L248.1,40.4 L222.4,40.4 L243.2,55.4 L235.3,79.6 L256.1,64.6 L268.9,55.4 L289.6,40.4 L289.6,40.4 L289.6,40.4 L289.6,40.4 Z"
          id="Shape3"
          fill="#FFFFFF"
          fillRule="nonzero"
        ></path>
        <path
          d="M360,64.7 L374.6,61 L380.7,79.8 L360,64.7 Z M393.6,40.4 L367.9,40.4 L360,16.2 L352.1,40.4 L326.4,40.4 L347.2,55.4 L339.3,79.6 L360.1,64.6 L372.9,55.4 L393.6,40.4 L393.6,40.4 L393.6,40.4 L393.6,40.4 Z"
          id="Shape4"
          fill="#FFFFFF"
          fillRule="nonzero"
        ></path>
        <path
          d="M464,64.7 L478.6,61 L484.7,79.8 L464,64.7 Z M497.6,40.4 L471.9,40.4 L464,16.2 L456.1,40.4 L430.4,40.4 L451.2,55.4 L443.3,79.6 L464.1,64.6 L476.9,55.4 L497.6,40.4 L497.6,40.4 L497.6,40.4 L497.6,40.4 Z"
          id="Shape5"
          fill="#FFFFFF"
          fillRule="nonzero"
        ></path>
      </g>
    </g>
  </g>
);
