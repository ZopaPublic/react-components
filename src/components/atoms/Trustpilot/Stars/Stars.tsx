import React, { SVGProps } from 'react';
import { FiveStar } from './ratings/FiveStar';
import { FourHalfStar } from './ratings/FourHalfStar';
import { FourStar } from './ratings/FourStar';
import { OneHalfStar } from './ratings/OneHalfStar';
import { OneStar } from './ratings/OneStar';
import { ThreeHalfStar } from './ratings/ThreeHalfStar';
import { ThreeStar } from './ratings/ThreeStar';
import { TwoHalfStar } from './ratings/TwoHalfStar';
import { TwoStar } from './ratings/TwoStar';
import { ZeroStar } from './ratings/ZeroStar';

export type TrustPilotStarsProps = SVGProps<SVGSVGElement> & {
  rating?: 0 | 1 | 1.5 | 2 | 2.5 | 3 | 3.5 | 4 | 4.5 | 5;
};

const trustPilotStarsLogoTitle = {
  0: 'Trustpilot Zero stars rating',
  1: 'Trustpilot One star',
  1.5: 'Trustpilot One and a half stars rating',
  2: 'Trustpilot Two stars rating',
  2.5: 'Trustpilot Two and a half stars rating',
  3: 'Trustpilot Three stars rating',
  3.5: 'Trustpilot Three and a half stars rating',
  4: 'Trustpilot Four stars rating',
  4.5: 'Trustpilot Four and a half stars rating',
  5: 'Trustpilot Five stars rating',
} as const;

const TrustPilotStars = ({ rating = 5, ...props }: TrustPilotStarsProps) => (
  <svg
    viewBox="0 0 512 96"
    height="20px"
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
    aria-labelledby="trust-pilot-rating-title trust-pilot-rating-desc"
    role="img"
    {...props}
  >
    <title id="trust-pilot-rating-title">{trustPilotStarsLogoTitle[rating]}</title>
    <desc id="trust-pilot-rating-desc">An illustration of the Trustpilot Stars Logo</desc>
    {rating === 0 && <ZeroStar />}
    {rating === 1 && <OneStar />}
    {rating === 1.5 && <OneHalfStar />}
    {rating === 2 && <TwoStar />}
    {rating === 2.5 && <TwoHalfStar />}
    {rating === 3 && <ThreeStar />}
    {rating === 3.5 && <ThreeHalfStar />}
    {rating === 4 && <FourStar />}
    {rating === 4.5 && <FourHalfStar />}
    {rating === 5 && <FiveStar />}
  </svg>
);

export default TrustPilotStars;
