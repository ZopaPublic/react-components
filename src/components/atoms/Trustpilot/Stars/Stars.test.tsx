import { render } from '@testing-library/react';
import axe from '../../../../../axe-helper';
import 'jest-styled-components';
import React from 'react';
import { FIVE_STAR_ID } from './ratings/FiveStar';
import { FOUR_HALF_STAR_ID } from './ratings/FourHalfStar';
import { FOUR_STAR_ID } from './ratings/FourStar';
import { ONE_HALF_STAR_ID } from './ratings/OneHalfStar';
import { ONE_STAR_ID } from './ratings/OneStar';
import { THREE_HALF_STAR_ID } from './ratings/ThreeHalfStar';
import { THREE_STAR_ID } from './ratings/ThreeStar';
import { TWO_HALF_STAR_ID } from './ratings/TwoHalfStar';
import { TWO_STAR_ID } from './ratings/TwoStar';
import { ZERO_STAR_ID } from './ratings/ZeroStar';
import Stars from './Stars';

describe('<TrustpilotStars />', () => {
  it('renders default 5 stars rating without a11y violations', async () => {
    const { container } = render(<Stars />);
    const results = await axe(container.innerHTML);
    expect(results).toHaveNoViolations();
    expect(container).toMatchSnapshot();
  });

  describe('Ratings', () => {
    const { container, rerender } = render(<Stars />);
    it.each`
      rating | ID
      ${0}   | ${ZERO_STAR_ID}
      ${1}   | ${ONE_STAR_ID}
      ${1.5} | ${ONE_HALF_STAR_ID}
      ${2}   | ${TWO_STAR_ID}
      ${2.5} | ${TWO_HALF_STAR_ID}
      ${3}   | ${THREE_STAR_ID}
      ${3.5} | ${THREE_HALF_STAR_ID}
      ${4}   | ${FOUR_STAR_ID}
      ${4.5} | ${FOUR_HALF_STAR_ID}
      ${5}   | ${FIVE_STAR_ID}
    `('renders the given rating: $rating', async ({ rating, ID }) => {
      rerender(<Stars rating={rating} />);
      expect(container.querySelector(`#${ID}`)).toBeTruthy();
    });
  });
});
