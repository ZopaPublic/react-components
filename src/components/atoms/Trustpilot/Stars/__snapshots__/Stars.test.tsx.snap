// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrustpilotStars /> renders default 5 stars rating without a11y violations 1`] = `
<div>
  <svg
    aria-labelledby="trust-pilot-rating-title trust-pilot-rating-desc"
    height="20px"
    role="img"
    viewBox="0 0 512 96"
    xmlns="http://www.w3.org/2000/svg"
    xmlns:xlink="http://www.w3.org/1999/xlink"
  >
    <title
      id="trust-pilot-rating-title"
    >
      Trustpilot Five stars rating
    </title>
    <desc
      id="trust-pilot-rating-desc"
    >
      An illustration of the Trustpilot Stars Logo
    </desc>
    <g
      fill="none"
      fill-rule="evenodd"
      id="Trustpilot_ratings_FiveStar"
      stroke="none"
      stroke-width="1"
    >
      <g>
        <g>
          <rect
            fill="#00B67A"
            fill-rule="nonzero"
            height="96"
            id="Rectangle-path1"
            width="96"
            x="0"
            y="0"
          />
          <rect
            fill="#00B67A"
            fill-rule="nonzero"
            height="96"
            id="Rectangle-path2"
            width="96"
            x="104"
            y="0"
          />
          <rect
            fill="#00B67A"
            fill-rule="nonzero"
            height="96"
            id="Rectangle-path3"
            width="96"
            x="208"
            y="0"
          />
          <rect
            fill="#00B67A"
            fill-rule="nonzero"
            height="96"
            id="Rectangle-path4"
            width="96"
            x="312"
            y="0"
          />
          <rect
            fill="#00B67A"
            fill-rule="nonzero"
            height="96"
            id="Rectangle-path5"
            width="96"
            x="416"
            y="0"
          />
          <path
            d="M48,64.7 L62.6,61 L68.7,79.8 L48,64.7 Z M81.6,40.4 L55.9,40.4 L48,16.2 L40.1,40.4 L14.4,40.4 L35.2,55.4 L27.3,79.6 L48.1,64.6 L60.9,55.4 L81.6,40.4 L81.6,40.4 L81.6,40.4 L81.6,40.4 Z"
            fill="#FFFFFF"
            fill-rule="nonzero"
            id="Shape1"
          />
          <path
            d="M152,64.7 L166.6,61 L172.7,79.8 L152,64.7 Z M185.6,40.4 L159.9,40.4 L152,16.2 L144.1,40.4 L118.4,40.4 L139.2,55.4 L131.3,79.6 L152.1,64.6 L164.9,55.4 L185.6,40.4 L185.6,40.4 L185.6,40.4 L185.6,40.4 Z"
            fill="#FFFFFF"
            fill-rule="nonzero"
            id="Shape2"
          />
          <path
            d="M256,64.7 L270.6,61 L276.7,79.8 L256,64.7 Z M289.6,40.4 L263.9,40.4 L256,16.2 L248.1,40.4 L222.4,40.4 L243.2,55.4 L235.3,79.6 L256.1,64.6 L268.9,55.4 L289.6,40.4 L289.6,40.4 L289.6,40.4 L289.6,40.4 Z"
            fill="#FFFFFF"
            fill-rule="nonzero"
            id="Shape3"
          />
          <path
            d="M360,64.7 L374.6,61 L380.7,79.8 L360,64.7 Z M393.6,40.4 L367.9,40.4 L360,16.2 L352.1,40.4 L326.4,40.4 L347.2,55.4 L339.3,79.6 L360.1,64.6 L372.9,55.4 L393.6,40.4 L393.6,40.4 L393.6,40.4 L393.6,40.4 Z"
            fill="#FFFFFF"
            fill-rule="nonzero"
            id="Shape4"
          />
          <path
            d="M464,64.7 L478.6,61 L484.7,79.8 L464,64.7 Z M497.6,40.4 L471.9,40.4 L464,16.2 L456.1,40.4 L430.4,40.4 L451.2,55.4 L443.3,79.6 L464.1,64.6 L476.9,55.4 L497.6,40.4 L497.6,40.4 L497.6,40.4 L497.6,40.4 Z"
            fill="#FFFFFF"
            fill-rule="nonzero"
            id="Shape5"
          />
        </g>
      </g>
    </g>
  </svg>
</div>
`;
